docs/testing.rst

```diff
@@ -0,0 +1,109 @@
+How is Raiden tested.
+#####################
+
+Raiden tests are split into these categories. These categories are ordered from
+the faster and easier to debug to the slower and harder to debug:
+
+- Unit tests: A collection of fast tests that do not require external services
+  to run. Because these tests don't require any services nor threads to run,
+  they don't have problems with flakyness and are easy to debug. These tests
+  are located under `raiden/tests/unit`.
+- Property tests: These tests use randomly generated events. These tests are
+  important ensure the system does not break when an unexpected series of
+  events happen. Because of the architecture used by the project these tests
+  are deterministic and do not require threads, therefore they are not flaky,
+  but can be harder to debug because of their random nature. These tests are
+  located under `raiden/tests/fuzz`.
+- Smoketest: This is a quick self contained test, that can run by user or
+  prior to the integration tests, to ensure the basics are working properly. It
+  will start a private chain, the necessary infrastructure for the transport
+  layer, deploy the smart contracts, start a Raiden instance and interact with
+  the blockchain. This test is available as a command line command, and it's
+  entry point is under `raiden/ui/cli.py`.
+- Integration tests: These are long running tests, that require external
+  services to run. Integration tests are slower to run, they have to start a
+  local private blockchain, a local transport infrastructure, multiple Raiden
+  nodes, and the REST API for these nodes. A fresh set of smart contracts are
+  deployed into this private chain, including Tokens and Raiden's smart
+  contracts. These tests are important to make sure Raiden will work as
+  expected with the Ethereum clients, however because they orchestrate multiple
+  systems they are considerably slower than the previous categories,
+  susceptible to flakyness, and harder to debug. These tests are located under
+  `raiden/tests/integration`.
+- Scenario tests: These tests run multiple times against the available
+  testnets. They exercise the Raiden nodes through the available REST API, in
+  the same a user application would. These tests are important to test the
+  system against a real life conditions (using the internet). These tests use
+  the scenario-player utility, and the scenarios are available under
+  `raiden/tests/scenarios`.
+
+Unit tests
```

Comment:
Why not put the above bullet points into those sections from the beginning? And have one section for each test type?